payments-core/src/main/java/com/stripe/android/payments/core/injection/AuthenticationModule.kt

```diff
@@ -0,0 +1,86 @@
+package com.stripe.android.payments.core.injection
+
+import androidx.activity.result.ActivityResultLauncher
+import com.stripe.android.Logger
+import com.stripe.android.PaymentAuthConfig
+import com.stripe.android.PaymentBrowserAuthStarter
+import com.stripe.android.PaymentRelayStarter
+import com.stripe.android.StripePaymentController
+import com.stripe.android.networking.AnalyticsRequestExecutor
+import com.stripe.android.networking.AnalyticsRequestFactory
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.payments.PaymentFlowResult
+import com.stripe.android.payments.core.authentication.NoOpIntentAuthenticator
+import com.stripe.android.payments.core.authentication.Stripe3DS2Authenticator
+import com.stripe.android.payments.core.authentication.WebIntentAuthenticator
+import com.stripe.android.stripe3ds2.service.StripeThreeDs2Service
+import com.stripe.android.stripe3ds2.transaction.MessageVersionRegistry
+import com.stripe.android.view.AuthActivityStarter
+import dagger.Module
+import dagger.Provides
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * [Module] for com.stripe.android.payments.core.authentication.
+ *
+ * All non-daggerized dependencies are passed in as constructor parameters for now, a parameter
+ * will be removed when it's daggerized and introduced to the dagger graph.
+ */
+@Module
+internal class AuthenticationModule(
+    private val stripeRepository: StripeRepository,
+    private val paymentRelayStarterFactory: (AuthActivityStarter.Host) -> PaymentRelayStarter,
+    private val paymentBrowserAuthStarterFactory: (AuthActivityStarter.Host) -> PaymentBrowserAuthStarter,
+    private val analyticsRequestExecutor: AnalyticsRequestExecutor,
+    private val analyticsRequestFactory: AnalyticsRequestFactory,
+    private val logger: Logger,
+    private val enableLogging: Boolean,
+    private val workContext: CoroutineContext,
+    private val uiContext: CoroutineContext,
+    private val threeDs2Service: StripeThreeDs2Service,
+    private val messageVersionRegistry: MessageVersionRegistry,
+    private val challengeProgressActivityStarter: StripePaymentController.ChallengeProgressActivityStarter,
+    private val stripe3ds2Config: PaymentAuthConfig.Stripe3ds2Config,
+    private val stripe3ds2ChallengeLauncher: ActivityResultLauncher<PaymentFlowResult.Unvalidated>?,
+) {
+    @Singleton
```

Comment:
What does it mean that this is `@Singleton` when `paymentRelayStarterFactory` is scoped to the Activity?

Reply:
hmm I don't think the factory is scoped to activity, rather it is also a singleton instance that returns a `PaymentRelayStarter.Modern` or `PaymentRelayStarter.Legacy` based on a `AuthActivityStarter.Host` passed to it.

When a `Stripe` or `FlowController` instance is created, the following chain would happen:
* one `StripePaymentController` instance is created within `Stripe ` or `FlowController`
* one `IntentAuthenticatorRegistry` instance is created within `StripePaymentController`
  * one instance of `paymentRelayStarterFactory` is created
  * the same instance of `paymentRelayStarterFactory` is used to create one instance of `NoOpIntentAuthenticator` and one instance of `Stripe3DS2Authenticator` 

The top scope above is at `Stripe` or `FlowController`, so there will be just once instance within this scope once the dep is marked`@Singleton`

A `AuthActivityStarter.Host` however is not created within this scope, instead it's created from an `Activity` or `Fragment` passed from public methods exposed in `Stripe` or `FlowController`, so it will not affect the scope of the `paymentRelayStarterFactory` here.

Reply:
Thanks for clearing that up.